Enable PUT requests on signed URLs

petergtz opened the pull request:
/cc @smoser-ibm 

* A short explanation of the proposed change:

    Extend the blobstore job to allow PUT requests on pre-signed URLs. This is a minimal change to discuss the usefulness of this feature. That means, the `/read/` path is obviously misleading when talking about a pre-signed URL used together with a PUT request, but made it possible to add this feature without much effort. To improve this, one could (a) either change that to say `/signed/`, or (b) introduce completely new pre-signed URLs using `/write/` which then also only allow PUT requests, while the `/read/` only allow GET/HEAD requests. To unblock us, the current PR is the most helpful though. Please advice.

* An explanation of the use cases your change solves

    We currently use the WebDAV blobstore as one possible backend for the bits-service. For the new protocol, it will be necessary to hand out a pre-signed URL to the client, to which it can PUT the bits. While we could roll our own WebDAV blobstore (by forking the one from CAPI), that would be counterproductive and could potentially introduce migration issues in the future.

* [x] I have viewed signed and have submitted the Contributor License Agreement

* [x] I have made this pull request to the `master` branch

* [x] I have run CF Acceptance Tests on bosh lite

aashah commented:
Hey Peter,

Thanks for the PR. Since completing this story: https://www.pivotaltracker.com/story/show/132781541, we have stopped committed directly to master. As such, can you switch the branch this pull request is targeting to develop. See: https://github.com/blog/2224-change-the-base-branch-of-a-pull-request

More relevant to the PR, we had two primary questions:

1. We wanted clarification on what 'pre-signed' means in this context. Our understanding is that these are **signed** URLs generated by CC.
1. By allowing PUT, everybody with read access will be granted with write access as well. We're trying to determine whether this has any negative consequences.

Thanks!
CAPI Community pair, @aashah @simonleung8

petergtz commented:
Hi @aashah, @simonleung8,

I changed the base branch to develop.

1. Sorry about the naming. I thought pre-signed and signed are synonymous more or less. I changed the title of the PR and I can do the same to the commit once everything else is sorted out. The mechanism for those signed URLs will stay the same. The change is really only that one can now PUT against these signed URLs that CC hands out to a client.

2. Yes, this is the main question. This is also why I suggested the alternative approach:

    > (b) introduce completely new pre-signed URLs using /write/ which then also only allow PUT requests, while the /read/ only allow GET/HEAD requests.

    Since the latter is more effort to implement, I first wanted to see, what you guys think. Once we agree on the solution, I can make that change. I think the main advantage of that latter suggestion is, that nothing will change as long as you don't hand out such `/write/` signed URLs.

Thanks,
Peter

/cc @smoser-ibm @suhlig
